plugins/pulp_rpm/plugins/importers/yum/associate.py

```diff
@@ -199,8 +207,14 @@ def copy_rpms(units, source_repo, dest_repo, import_conduit, copy_deps, solver=N
     for unit in units:
         # we are passing in units that may have flattened "provides" metadata.
         # This flattened field is not used by associate_single_unit().
-        repo_controller.associate_single_unit(dest_repo, unit)
-        unit_set.add(unit)
+        if rpm_parse.signature_enabled(config):
```

Comment:
Consider rearranging this to

```
    for unit in units:
        if rpm_parse.signature_enabled(config):
            if unit.downloaded:
                rpm_parse.verify_signature(unit, config)
            else:
                continue
        repo_controller.associate_single_unit(repository=dest_repo, unit=unit)
        unit_set.add(unit)
```

I find that a bit easier to read. It eliminates the need for two copies of the last two lines. The ``continue`` line also makes it clear that you are deliberately not associating the unit in that case.

But it is also fine how you have it if you prefer it that way.